internal/pkg/service/stream/storage/statistics/repository/stats_delete_test.go

```diff
@@ -0,0 +1,315 @@
+	client := mocked.TestEtcdClient()
+	defRepo := d.DefinitionRepository()
+	storageRepo := d.StorageRepository()
+	fileRepo := storageRepo.File()
+	sliceRepo := storageRepo.Slice()
+	volumeRepo := storageRepo.Volume()
+	statsRepo := d.StatisticsRepository()
+
+	// Log etcd operations
+	var etcdLogs bytes.Buffer
+	rawClient := d.EtcdClient()
+	rawClient.KV = etcdlogger.KVLogWrapper(rawClient.KV, &etcdLogs, etcdlogger.WithMinimal())
+
+	// Register active volumes
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		session, err := concurrency.NewSession(client)
+		require.NoError(t, err)
+		defer func() { require.NoError(t, session.Close()) }()
+		test.RegisterWriterVolumes(t, ctx, volumeRepo, session, 1)
+	}
+
+	// Create branch, source, sink, file, slice
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		branch := test.NewBranch(branchKey)
+		require.NoError(t, defRepo.Branch().Create(&branch, clk.Now(), by).Do(ctx).Err())
+		source := test.NewSource(sourceKey)
+		require.NoError(t, defRepo.Source().Create(&source, clk.Now(), by, "Create source").Do(ctx).Err())
+		sink := test.NewSinkWithLocalStorage(sinkKey)
+		require.NoError(t, defRepo.Sink().Create(&sink, clk.Now(), by, "Create sink").Do(ctx).Err())
+	}
+
+	// Create 2 more files/slices (3 totally)
+	// -----------------------------------------------------------------------------------------------------------------
+	var fileKey1, fileKey2, fileKey3 model.FileKey
+	var sliceKey1, sliceKey2, sliceKey3 model.SliceKey
+	{
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.Rotate(sinkKey, clk.Now()).Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.Rotate(sinkKey, clk.Now()).Do(ctx).Err())
+
+		files, err := fileRepo.ListIn(sinkKey).Do(ctx).All()
+		require.NoError(t, err)
+		require.Len(t, files, 3)
+		fileKey1 = files[0].FileKey
+		fileKey2 = files[1].FileKey
+		fileKey3 = files[2].FileKey
+
+		slices, err := sliceRepo.ListIn(sinkKey).Do(ctx).All()
+		require.NoError(t, err)
+		require.Len(t, slices, 3)
+		sliceKey1 = slices[0].SliceKey
+		sliceKey2 = slices[1].SliceKey
+		sliceKey3 = slices[2].SliceKey
+	}
+
+	// Create stats records
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		value := statistics.Value{
+			SlicesCount:      1,
+			FirstRecordAt:    utctime.MustParse("2000-01-01T01:00:00.000Z"),
+			LastRecordAt:     utctime.MustParse("2000-01-01T02:00:00.000Z"),
+			RecordsCount:     1,
+			UncompressedSize: 1,
+			CompressedSize:   1,
+		}
+		assert.NoError(t, statsRepo.Put(ctx, []statistics.PerSlice{
+			{SliceKey: sliceKey1, Value: value},
+			{SliceKey: sliceKey2, Value: value},
+			{SliceKey: sliceKey3, Value: value},
+		}))
+	}
+
+	// Move slices 1 and 2 to the storage.Staging, it moves also related statistics
```

Comment:
Maybe we could document what `storage.Staging` and `storage.Target` means?
We have multiple ways how to call specific state of slice/file in comparison to storage level it is in

Reply:
It is already documented 
![image](https://github.com/keboola/keboola-as-code/assets/19371734/a3c5cdfd-695f-4429-acf9-59ba721e2691)

Reply:
See [FileState.ToLevel](https://github.com/keboola/keboola-as-code/blob/f9413c4782852cef5909dce70146b34c7dc6156e/internal/pkg/service/stream/storage/model/filestate.go#L74-L86) and [SliceState.ToLevel](https://github.com/keboola/keboola-as-code/blob/f9413c4782852cef5909dce70146b34c7dc6156e/internal/pkg/service/stream/storage/model/slicestate.go#L62-L74).